cni/pkg/ambient/server.go

```diff
@@ -80,17 +84,26 @@ func NewServer(ctx context.Context, args AmbientArgs) (*Server, error) {
 		marshalableDisabledSelectors: ambientpod.LegacyLabelSelector,
 		kubeClient:                   client,
 	}
-	s.iptablesCommand = lazy.New(func() (string, error) {
-		return s.detectIptablesCommand(), nil
-	})
 
-	// We need to find our Host IP -- is there a better way to do this?
-	h, err := GetHostIP(s.kubeClient.Kube())
-	if err != nil || h == "" {
-		return nil, fmt.Errorf("error getting host IP: %v", err)
+	switch args.RedirectMode {
+	case IptablesMode:
+		s.redirectMode = IptablesMode
+		s.iptablesCommand = lazy.New(func() (string, error) {
+			return s.detectIptablesCommand(), nil
+		})
+
+		// We need to find our Host IP -- is there a better way to do this?
+		h, err := GetHostIP(s.kubeClient.Kube())
+		if err != nil || h == "" {
+			return nil, fmt.Errorf("error getting host IP: %v", err)
+		}
+		HostIP = h
+		log.Infof("HostIP=%v", HostIP)
+	case EbpfMode:
+		s.redirectMode = EbpfMode
+		s.ebpfServer = ebpf.NewRedirectServer()
+		s.ebpfServer.Start(ctx.Done())
```

Comment:
nit: make Start not return anything since its always `nil`

Reply:
fixed